### PR TITLE
A0-4251: Move `sign_with_params` to a new trait in aleph client

### PR DESCRIPTION
**Pull request by ggawryal**
# Description

In #1602, we've added `sign_with_params` to `SignedConnectionApi` in aleph-client to be able to specify nonce in lower level api for signing transactions, without submitting. Normally, this could be done in synchronous manner.
However, in Most bridge we hide signing keys in a separate enclave, and implement `SignedConnectionApi` by a client talking to that enclave, to be a proxy between that enclave and a user. This:
1. Complicates creating synchronous signing implementation for that proxy, as a channel to the enclave is async,
2. Forces implementing that lower level method, when it is in fact not needed in Most use case.

Therefore, I'm splitting that trait.
Also, in that PR I forgot to make `SubmittableExtrinsic` public, so I'm doing it now.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- I have made corresponding changes to the existing documentation
- I have bumped `aleph-client` version

### CODE REVIEW

**Review by obrok — APPROVED** [review]
Code itself looks good to me. I guess, the question is. if we want to do _any_ changes directed at making `aleph_client` more usable when in fact it should not be used (see discussion on slack).

File: aleph-client/src/connections.rs [hunk 1]
```diff
@@ -183,6 +183,18 @@ pub trait SignedConnectionApi: ConnectionApi {
...
     ) -> anyhow::Result<TxInfo>;
 
+    /// Returns account id which signs this connection
+    fn account_id(&self) -> &AccountId;
+
+    /// Returns a [`KeyPair`] which signs this connection
+    fn signer(&self) -> &KeyPair;
+
+    /// Tries to convert [`SignedConnection`] as [`RootConnection`]
+    async fn try_as_root(&self) -> anyhow::Result<RootConnection>;
```
Comment (kostekIV) [inline aleph-client/src/connections.rs:193]: actually lets add a default impl where we return an error like `not upgradable to root` or smth

**Review by kostekIV — APPROVED** [review]
lgtm

File: aleph-client/src/connections.rs (hunk 1 above)
Comment (ggawryal) [reply aleph-client/src/connections.rs]: done

**Pull request merged**